gulpfile.coffee

```diff
@@ -1,69 +1,217 @@
-gulp = require('gulp')
-$ = require('gulp-load-plugins')()
+fs = require 'fs'
+gulp = require('gulp-help')(require('gulp'))
+
 main_bower_files = require 'main-bower-files'
 del = require 'del'
 exec = require('child_process').exec
 minimist = require 'minimist'
 
+$ = require('gulp-load-plugins')()
+
+
+dir_bower_components = 'bower_components'
+dir_main = 'main'
+dir_node_modules = 'node_modules'
+
+dir_lib = "#{dir_main}/lib"
+dir_libx = "#{dir_main}/libx"
+dir_temp = 'temp'
+dir_venv = "#{dir_temp}/venv"
+
+file_lib = "#{dir_lib}.zip"
+
+dir_static = "#{dir_main}/static"
+dir_src = "#{dir_static}/src"
+dir_script = "#{dir_src}/script"
+dir_style = "#{dir_src}/style"
+dir_ext = "#{dir_static}/ext"
+dir_min = "#{dir_static}/min"
+dir_storage = "#{dir_temp}/storage"
+
 
-root_dir = './main'
-static_dir = "#{root_dir}/static"
 paths =
+  ext: [
+      "#{dir_ext}/jquery/dist/jquery.js"
+      "#{dir_ext}/moment/moment.js"
+      "#{dir_ext}/nprogress/nprogress.js"
+      "#{dir_ext}/bootstrap-sweetalert/lib/sweet-alert.js"
+      "#{dir_ext}/bootstrap/js/alert.js"
+      "#{dir_ext}/bootstrap/js/button.js"
+      "#{dir_ext}/bootstrap/js/transition.js"
+      "#{dir_ext}/bootstrap/js/collapse.js"
+      "#{dir_ext}/bootstrap/js/dropdown.js"
+      "#{dir_ext}/bootstrap/js/tooltip.js"
+    ]
   clean: [
-      "#{static_dir}/dst"
-      "#{static_dir}/ext"
-      "#{static_dir}/min"
+      dir_min
+      './**/*.pyc'
+      './**/*.pyo'
+      './**/*.~'
+    ]
+  clean_all: [
+      dir_min
+      dir_bower_components
+      dir_node_modules
+      "#{dir_temp}/pip.guard"
+      "#{dir_temp}/npm.guard"
+      "#{dir_temp}/bower.guard"
+    ]
+  clean_py: [
+      dir_lib
+      dir_venv
+    ]
+  style: [
+      "#{dir_style}/style.less"
+    ]
+  script: [
+      "#{dir_script}/**/*.coffee"
```

Comment:
That is quite nice.. as you don't want to change anything because usually it will work..